src/code/z_parameter.c

```diff
@@ -2564,49 +2873,573 @@ s16 D_801BFC40[] = {
+        }
+    }
+
+    // Update letter colors
+    if (interfaceCtx->minigamePerfectLetterCount == MINIGAME_PERFECT_NUM_LETTERS) {
+        if (interfaceCtx->minigamePerfectState[MINIGAME_PERFECT_NUM_LETTERS - 1] == MINIGAME_PERFECT_STATE_DISPLAY) {
+
+            colorStepR =
+                ABS_ALT(interfaceCtx->minigamePerfectPrimColor[0] -
+                        sMinigamePerfectType2PrimColorTargets[interfaceCtx->minigamePerfectColorTargetIndex][0]) /
+                interfaceCtx->minigamePerfectColorTimer;
+            colorStepG =
+                ABS_ALT(interfaceCtx->minigamePerfectPrimColor[1] -
+                        sMinigamePerfectType2PrimColorTargets[interfaceCtx->minigamePerfectColorTargetIndex][1]) /
+                interfaceCtx->minigamePerfectColorTimer;
+            colorStepB =
+                ABS_ALT(interfaceCtx->minigamePerfectPrimColor[2] -
+                        sMinigamePerfectType2PrimColorTargets[interfaceCtx->minigamePerfectColorTargetIndex][2]) /
+                interfaceCtx->minigamePerfectColorTimer;
+
+            if (interfaceCtx->minigamePerfectPrimColor[0] >=
+                sMinigamePerfectType2PrimColorTargets[interfaceCtx->minigamePerfectColorTargetIndex][0]) {
+                interfaceCtx->minigamePerfectPrimColor[0] -= colorStepR;
+            } else {
+                interfaceCtx->minigamePerfectPrimColor[0] += colorStepR;
+            }
+
+            if (interfaceCtx->minigamePerfectPrimColor[1] >=
+                sMinigamePerfectType2PrimColorTargets[interfaceCtx->minigamePerfectColorTargetIndex][1]) {
+                interfaceCtx->minigamePerfectPrimColor[1] -= colorStepG;
+            } else {
+                interfaceCtx->minigamePerfectPrimColor[1] += colorStepG;
+            }
+
+            if (interfaceCtx->minigamePerfectPrimColor[2] >=
+                sMinigamePerfectType2PrimColorTargets[interfaceCtx->minigamePerfectColorTargetIndex][2]) {
+                interfaceCtx->minigamePerfectPrimColor[2] -= colorStepB;
+            } else {
+                interfaceCtx->minigamePerfectPrimColor[2] += colorStepB;
+            }
+
+            interfaceCtx->minigamePerfectColorTimer--;
+            if (interfaceCtx->minigamePerfectColorTimer == 0) {
+                interfaceCtx->minigamePerfectColorTimer = 20;
+                interfaceCtx->minigamePerfectColorTargetIndex ^= 1;
+                interfaceCtx->minigamePerfectTimer++;
+                if (interfaceCtx->minigamePerfectTimer == 6) {
+                    for (i = 0; i < MINIGAME_PERFECT_NUM_LETTERS; i++) {
+                        interfaceCtx->minigamePerfectState[i] = MINIGAME_PERFECT_STATE_EXIT;
+                    }
+                    interfaceCtx->minigamePerfectColorTimer = 5;
+                }
+            }
+        }
+    }
+}
+
+// Targets to offset each letter to properly spell "PERFECT!"
+s16 sMinigamePerfectType3LetterXOffsetSpellingTargets[MINIGAME_PERFECT_NUM_LETTERS] = {
```

Comment:
I feel like you lose very little information by doing `sMinigamePerfectType3TargetXs` or something similar, the comment does most of the rest. Applies to other names too.

Reply:
The tricky part about this comes from Type 2:
`sMinigamePerfectType2LetterXOffsetSpellingTargets`
from
`sMinigamePerfectType2LetterXOffsetOffScreenTargets`

Both are x-offset targets that apply to the vtx Offset that creates the effect of spreading the letters out horizontally. 

However, the first one is to go from all 8 letters overlapping in the center of the screen -> spread out horizontally to nicely spelling out (PERFECT!).

And the second one goes from the letters nicely spelling out (PERFECT!) -> spread out horizontally to offscreen as a fading effect

Reply:
Yeah, that is an issue, hmm. `Letter` is at least redundant, and I think we can still drop `Offset` with a comment, though.